plugins/modules/database/influxdb/influxdb_retention_policy.py

```diff
@@ -156,12 +166,25 @@ def alter_retention_policy(module, client, retention_policy):
     elif duration == 'INF':
         influxdb_duration_format = '0'
 
+    if shard_group_duration:
+        shard_group_duration_lookup = duration_regexp.search(shard_group_duration)
+        if shard_group_duration_lookup.group(2) == 'h':
```

Comment:
```suggestion
		if not shard_group_duration:
			module.fail_json(msg="Failed to parse value of shard_group_duration. Please see the documentation for valid values")
        if shard_group_duration_lookup.group(2) == 'h':
```

Reply:
This is to avoid - 

```
>>> shard_group_duration_lookup = duration_regexp.search('aaa')
>>> shard_group_duration_lookup
>>> shard_group_duration_lookup.group(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'group'
```